code/modules/modular_computers/ntos/components.dm

```diff
@@ -33,6 +33,17 @@
 	else
 		return 0
 
+/datum/extension/interactive/ntos/proc/get_network_speed(var/specific_action = 0)
+	switch(get_ntnet_status(NTNET_SOFTWAREDOWNLOAD))
+		if(0)
+			return 0
+		if(1)
+			return NTNETSPEED_LOWSIGNAL
+		if(2)
+			return NTNETSPEED_HIGHSIGNAL
+		if(3)
+			return NTNETSPEED_ETHERNET
```

Comment:
Is there a reason it doesn't just always use speed defines? I think I made a comment above based on a misunderstanding of the relationship between these numbers and these defines.

Reply:
remains of / setup for signal strength stuff, connection type is used as sole factor because im too lazy to add that now.